src/search.cpp

```diff
@@ -1250,6 +1250,11 @@ namespace {
       {
           Depth r = reduction<PvNode>(improving, depth, moveCount);
 
+#ifdef ANTI
+          if (pos.is_anti() && pos.can_capture())
+              r -= r ? ONE_PLY : DEPTH_ZERO;
+          else
```

Comment:
On the one hand I understand your motivation to test variations of the patch to better understand why it works, on the other hand I am a bit confused about the reasoning behind the requirements for accepting the PR. Accepting the PR only if you find a patch that replaces the PR is basically the same as rejecting the PR. Furthermore, I think that the current implementation of SEE for giveaway has no clear interpretation, so I am not sure whether a variation there can help to understand changes. https://github.com/ddugovic/Stockfish/compare/492493e...c50e225 makes much more sense to me for comparison purposes, even though comparing it to the simplification only might give more information if you are only interested in the effect of the `if (pos.can_capture()) ...`.

> as the context surrounding the code change (including the immediately following else block) has me concerned about the possibility of this and/or other changes overfitting (to a local maximum Elo), a situation which is difficult to remedy.

Yes, overfitting should be avoided, but that is one thing simplification tests are for. The change does not add tons of special cases, so I think it should not be that problematic. However, LMR changes can be very sensitive to TC, so one should keep an eye on that.

In my opinion, the basic idea of a patch should be understandable (something like "less reduction if white has exactly 5 pawns" wouldn't, IMO), and test results should be fine, but I think you can not expect to exactly understand why one patch works and the other doesn't, since there is just too much interference between different parts of search, evaluation, move ordering, etc., that can lead to unintuitive results in modern chess engines. Official stockfish does not seem to have run into a dead end with its massive focus on test results.

> Normally I wouldn't hesitate to merge a PR, but here I can't guarantee that I can improve an entire +5 Elo over anti_lmr5; maybe I can only improve +9 (or fewer) Elo over master with a similar patch, because I lacked the foresight to provide feedback on code I overlooked prior to LTC.

This is a a way of thinking I completely disagree with. Fearing not to be able to improve further should not be a reason to keep you from improving, and by the way, why it is important that you improve over it with a similar patch?

If you think that the testing methodology does not fit its purpose, it can be discussed, but that is in my opinion not in any way related to this PR.